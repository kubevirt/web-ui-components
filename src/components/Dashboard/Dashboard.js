import React from 'react';
import PropTypes from 'prop-types';

const Dashboard = ({ children }) => <div>{children}</div>;

Dashboard.defaultProps = {
  children: null,
};

Dashboard.propTypes = {
  children: PropTypes.node,
};

export default Dashboard;
