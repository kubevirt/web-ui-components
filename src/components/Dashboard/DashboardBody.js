import React from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

const DashboardBody = ({ className, children }) => (
  <div className={classNames('kubevirt-dashboard__body', className)}>{children}</div>
);

DashboardBody.defaultProps = {
  className: null,
  children: null,
};

DashboardBody.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
};

export default DashboardBody;
