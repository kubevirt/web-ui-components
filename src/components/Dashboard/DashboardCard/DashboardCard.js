import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Card } from 'patternfly-react';

const DashboardCard = ({ className, children, ...props }) => (
  <Card {...props} className={classNames('kubevirt-dashboard__card', className)}>
    {children}
  </Card>
);

DashboardCard.defaultProps = {
  className: null,
  children: null,
};

DashboardCard.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

export default DashboardCard;
