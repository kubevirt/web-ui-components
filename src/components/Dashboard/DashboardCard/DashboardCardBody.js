import React from 'react';
import PropTypes from 'prop-types';

import { CardBody } from 'patternfly-react';

import { Loading } from '../../Loading';

class DashboardCardBody extends React.PureComponent {
  render() {
    const { LoadingComponent, isLoading, children, ...props } = this.props;
    return <CardBody {...props}>{isLoading ? <LoadingComponent /> : children}</CardBody>;
  }
}

DashboardCardBody.defaultProps = {
  LoadingComponent: Loading,
  isLoading: false,
  children: null,
};

DashboardCardBody.propTypes = {
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  isLoading: PropTypes.bool,
  children: PropTypes.node,
};

export default DashboardCardBody;
