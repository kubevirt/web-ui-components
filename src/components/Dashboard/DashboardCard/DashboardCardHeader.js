import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { CardHeading } from 'patternfly-react';

class DashboardCardHeader extends React.PureComponent {
  render() {
    const { className, children, ...props } = this.props;
    return (
      <CardHeading {...props} className={classNames('kubevirt-dashboard__card-heading', className)}>
        {children}
      </CardHeading>
    );
  }
}

DashboardCardHeader.defaultProps = {
  className: null,
  children: null,
};

DashboardCardHeader.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

export default DashboardCardHeader;
