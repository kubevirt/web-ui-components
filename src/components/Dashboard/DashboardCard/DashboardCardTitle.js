import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { CardTitle } from 'patternfly-react';

class DashboardCardTitle extends React.PureComponent {
  render() {
    const { className, children, ...props } = this.props;
    return (
      <CardTitle {...props} className={classNames('kubevirt-dashboard__card-title', className)}>
        {children}
      </CardTitle>
    );
  }
}

DashboardCardTitle.defaultProps = {
  children: null,
  className: null,
};

DashboardCardTitle.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

export default DashboardCardTitle;
