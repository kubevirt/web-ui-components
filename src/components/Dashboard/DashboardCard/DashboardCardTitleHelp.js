import React from 'react';
import PropTypes from 'prop-types';
import { Button, Icon, OverlayTrigger, Popover } from 'patternfly-react';

class DashboardCardTitleHelp extends React.PureComponent {
  render() {
    const { children } = this.props;
    if (React.Children.count(children) === 0) {
      return null;
    }
    const overlay = <Popover id="popover">{children}</Popover>;
    return (
      <OverlayTrigger overlay={overlay} placement="top" trigger={['click']} rootClose>
        <Button bsStyle="link">
          <Icon
            type="fa"
            name="info-circle"
            className="kubevirt-dashboard__icon-sm kubevirt-dashboard__heading-icon--info"
          />
        </Button>
      </OverlayTrigger>
    );
  }
}

DashboardCardTitleHelp.defaultProps = {
  children: null,
};

DashboardCardTitleHelp.propTypes = {
  children: PropTypes.node,
};

export default DashboardCardTitleHelp;
