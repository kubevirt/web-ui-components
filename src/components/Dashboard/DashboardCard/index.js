export { default as DashboardCard } from './DashboardCard';
export { default as DashboardCardBody } from './DashboardCardBody';
export { default as DashboardCardHeader } from './DashboardCardHeader';
export { default as DashboardCardTitle } from './DashboardCardTitle';
export { default as DashboardCardTitleHelp } from './DashboardCardTitleHelp';
export { DashboardCardTitleSeeAll } from './DashboardCardTitleSeeAll';
