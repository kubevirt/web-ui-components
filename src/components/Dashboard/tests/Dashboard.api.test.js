import React from 'react';
import { mount } from 'enzyme';

import Dashboard from '../Dashboard';
import DashboardBody from '../DashboardBody';
import DashboardCard from '../DashboardCard/DashboardCard';
import DashboardCardHeader from '../DashboardCard/DashboardCardHeader';
import DashboardCardTitle from '../DashboardCard/DashboardCardTitle';
import DashboardCardTitleHelp from '../DashboardCard/DashboardCardTitleHelp';
import DashboardCardBody from '../DashboardCard/DashboardCardBody';

const testDashboard = () => (
  <Dashboard>
    <DashboardBody>
      <DashboardCard>
        <DashboardCardHeader>
          <DashboardCardTitle>Red Card</DashboardCardTitle>
          <DashboardCardTitleHelp>Red Card help</DashboardCardTitleHelp>
        </DashboardCardHeader>
        <DashboardCardBody isLoading>Red card content</DashboardCardBody>
      </DashboardCard>
      <DashboardCard>
        <DashboardCardHeader>
          <DashboardCardTitle>Blue Card</DashboardCardTitle>
        </DashboardCardHeader>
        <DashboardCardBody>Blue card content</DashboardCardBody>
      </DashboardCard>
    </DashboardBody>
  </Dashboard>
);

describe('<Dashboard /> API', () => {
  it('renders correctly', () => {
    const component = mount(testDashboard());
    expect(component).toMatchSnapshot();
  });
});
