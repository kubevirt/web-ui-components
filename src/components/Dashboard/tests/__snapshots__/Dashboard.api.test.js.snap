// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> API renders correctly 1`] = `
<Dashboard>
  <div>
    <DashboardBody
      className={null}
    >
      <div
        className="kubevirt-dashboard__body"
      >
        <DashboardCard
          className={null}
        >
          <Card
            accented={false}
            aggregated={false}
            aggregatedMini={false}
            cardRef={null}
            className="kubevirt-dashboard__card"
            matchHeight={false}
          >
            <div
              className="card-pf kubevirt-dashboard__card"
            >
              <DashboardCardHeader
                className={null}
              >
                <CardHeading
                  className="kubevirt-dashboard__card-heading"
                >
                  <div
                    className="card-pf-heading kubevirt-dashboard__card-heading"
                  >
                    <DashboardCardTitle
                      className={null}
                    >
                      <CardTitle
                        className="kubevirt-dashboard__card-title"
                      >
                        <h2
                          className="card-pf-title kubevirt-dashboard__card-title"
                        >
                          Red Card
                        </h2>
                      </CardTitle>
                    </DashboardCardTitle>
                    <DashboardCardTitleHelp>
                      <OverlayTrigger
                        defaultOverlayShown={false}
                        overlay={
                          <Popover
                            bsClass="popover"
                            id="popover"
                            placement="right"
                          >
                            Red Card help
                          </Popover>
                        }
                        placement="top"
                        rootClose={true}
                        trigger={
                          Array [
                            "click",
                          ]
                        }
                      >
                        <Button
                          active={false}
                          block={false}
                          bsClass="btn"
                          bsStyle="link"
                          disabled={false}
                          onClick={[Function]}
                        >
                          <button
                            className="btn btn-link"
                            disabled={false}
                            onClick={[Function]}
                            type="button"
                          >
                            <Icon
                              className="kubevirt-dashboard__icon-sm kubevirt-dashboard__heading-icon--info"
                              name="info-circle"
                              type="fa"
                            >
                              <FontAwesome
                                className="kubevirt-dashboard__icon-sm kubevirt-dashboard__heading-icon--info"
                                name="info-circle"
                              >
                                <span
                                  aria-hidden={true}
                                  className="fa fa-info-circle kubevirt-dashboard__icon-sm kubevirt-dashboard__heading-icon--info"
                                />
                              </FontAwesome>
                            </Icon>
                          </button>
                        </Button>
                      </OverlayTrigger>
                    </DashboardCardTitleHelp>
                  </div>
                </CardHeading>
              </DashboardCardHeader>
              <DashboardCardBody
                LoadingComponent={[Function]}
                isLoading={true}
              >
                <CardBody
                  className=""
                >
                  <div
                    className="card-pf-body"
                  >
                    <Loading
                      text="Loading"
                    >
                      <div
                        className="blank-slate-pf"
                        key="progress"
                      >
                        <div
                          className="spinner spinner-lg blank-slate-pf-icon"
                        />
                        <h3
                          className="blank-slate-pf-main-action"
                        >
                          Loading
                        </h3>
                      </div>
                    </Loading>
                  </div>
                </CardBody>
              </DashboardCardBody>
            </div>
          </Card>
        </DashboardCard>
        <DashboardCard
          className={null}
        >
          <Card
            accented={false}
            aggregated={false}
            aggregatedMini={false}
            cardRef={null}
            className="kubevirt-dashboard__card"
            matchHeight={false}
          >
            <div
              className="card-pf kubevirt-dashboard__card"
            >
              <DashboardCardHeader
                className={null}
              >
                <CardHeading
                  className="kubevirt-dashboard__card-heading"
                >
                  <div
                    className="card-pf-heading kubevirt-dashboard__card-heading"
                  >
                    <DashboardCardTitle
                      className={null}
                    >
                      <CardTitle
                        className="kubevirt-dashboard__card-title"
                      >
                        <h2
                          className="card-pf-title kubevirt-dashboard__card-title"
                        >
                          Blue Card
                        </h2>
                      </CardTitle>
                    </DashboardCardTitle>
                  </div>
                </CardHeading>
              </DashboardCardHeader>
              <DashboardCardBody
                LoadingComponent={[Function]}
                isLoading={false}
              >
                <CardBody
                  className=""
                >
                  <div
                    className="card-pf-body"
                  >
                    Blue card content
                  </div>
                </CardBody>
              </DashboardCardBody>
            </div>
          </Card>
        </DashboardCard>
      </div>
    </DashboardBody>
  </div>
</Dashboard>
`;
